batchflow/models/tf/mobilenet.py

```diff
@@ -189,38 +229,114 @@ def body(cls, inputs, name='body', **kwargs):
             whether to make a residual connection
         strides : int
             stride for 3x3 convolution
         expansion_factor : int
             multiplier for the number of filters in internal convolutions
         width_factor : float
             multiplier for the number of filters
+        activation : callable, optional
+            If not specified tf.nn.relu is used.
+        se_block : bool or dict
+            whether to include squeeze and excitation block
+            If dict, it must contain :meth:`~TFModel.se_block` params
         name : str
             scope name
 
         Returns
         -------
         tf.Tensor
         """
+
         num_filters = int(cls.num_channels(inputs, kwargs.get('data_format')) * expansion_factor * width_factor)
-        conv_filters = [num_filters, num_filters, filters]
-        x = conv_block(inputs, 'cna Cna cn', conv_filters, [1, 3, 1], name=name, strides=[1, strides, 1], **kwargs)
+        x = conv_block(inputs, 'cnaWna', [num_filters, num_filters], [1, kernel_size], strides=[1, strides],
+                       name='%s-exp' % name, **kwargs)
+
+        if se_block:
+            if not isinstance(se_block, dict):
+                se_block = dict(activation=[kwargs.get('activation', tf.nn.relu), h_sigmoid], ratio=num_filters // 4)
+            se_block = {**kwargs, **se_block}
+            x = cls.se_block(x, name='%s-se' % name, **se_block)
+        x = conv_block(x, 'cn', filters, 1, name='%s-down' % name, **kwargs)
+
         if residual:
             x = inputs + x
         return x
+
+
+class MobileNet_v3(MobileNet_v2):
+    """ MobileNet version 3 architecture """
+    @classmethod
+    def default_config(cls):
+        config = TFModel.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_LARGE_DEFAULT_BODY)))
+        config['head'].update(dict(layout='cnavcacV', filters=[960, 1280, 2], pool_size=7, kernel_size=1,
+                                   activation=h_swish))
+        config['loss'] = 'ce'
+        return config
+
+
+class MobileNet_v3_small(MobileNet_v2):
+    """ MobileNet version 3 small architecture """
+    @classmethod
+    def default_config(cls):
+        config = TFModel.default_config()
+        config['initial_block'].update(dict(layout='cna', filters=16, kernel_size=3, strides=2, activation=h_swish))
+        config['body'].update(dict(width_factor=1, layout=deepcopy(_V3_SMALL_DEFAULT_BODY)))
+        config['head'].update(dict(layout='cnavcacV', filters=[576, 1280, 2], pool_size=7,
+                                   kernel_size=1, activation=h_swish,
+                                   se_block=dict(activation=[h_swish, h_sigmoid], ratio=144)))
+        config['loss'] = 'ce'
+        return config
+
+    @classmethod
+    def head(cls, inputs, se_block=None, name='head', **kwargs):
```

Comment:
Move it to MobileNet_v3, as according to the article head differs by config only, not by structure. MobileNet v3 Large could have se block in the head as well

Reply:
done.